girder/models/upload.py

```diff
@@ -172,10 +175,13 @@ def finalizeUpload(self, upload, assetstore=None):
         self.remove(upload)
 
         # Add an async event for handlers that wish to process this file.
-        events.daemon.trigger('data.process', {
+        eventParams = {
             'file': file,
             'assetstore': assetstore
-        })
+        }
+        if 'reference' in upload:
+            eventParams['reference'] = upload['reference']
+        events.daemon.trigger('data.process', eventParams)
```

Comment:
Hey @manthey,
FIY, we also have a pull request to reference the upload object here, see https://github.com/girder/girder/pull/1236